packages/spacecat-shared-rum-api-client/src/functions/opportunities.js

```diff
@@ -69,11 +69,19 @@ function handler(bundles) {
   //     data[bundle.url][weekKey].metrics.push({ selector: source, ctr });
   // }
     // Initialize an object to hold the selectors and their counts
-    let globalSelectors = {};
+    const globalSelectors = {};
 
     for (const bundle of bundles) {
       // Initialize a Set to hold the unique selectors for this bundle
-      let uniqueSelectors = new Set();
+      const uniqueSelectors = new Set();
```

Comment:
I'd move the declaration outside of the loop so you re-use the same variable and avoid allocation of hundreds of different ones.
```js
    let uniqueSelectors;
    for (const bundle of bundles) {
      // Initialize a Set to hold the unique selectors for this bundle
      uniqueSelectors = new Set();
```